File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        for (int num : nums) {
            freq[num]++;
        }

        vector<int> res;
        for (auto& it : freq) {
            res.push_back(it.first);
        }

        // Sort by frequency (descending)
        sort(res.begin(), res.end(), [&](int a, int b) {
            return freq[a] > freq[b];
        });

        // Return top k elements
        return vector<int>(res.begin(), res.begin() + k);
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();

        // Count frequencies
        for (int num : nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        // Sort keys by frequency (descending)
        List<Integer> unique = new ArrayList<>(freq.keySet());
        unique.sort((a, b) -> freq.get(b) - freq.get(a));

        // Pick top k elements
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = unique.get(i);
        }

        return result;
    }
}
